HeterogeneousCore/CUDAUtilities/interface/prefixScan.h

```diff
@@ -129,13 +129,14 @@ namespace cms {
 
     // limited to 1024*1024 elements....
     template <typename T>
-    __global__ void multiBlockPrefixScan(T const* __restrict__ ci, T* __restrict__ co, int32_t size, int32_t* pc) {
+    __global__ void multiBlockPrefixScan(T const* ci, T* co, int32_t size, int32_t* pc) {
       __shared__ T ws[32];
       // first each block does a scan of size 1024; (better be enough blocks....)
-      assert(1024 * gridDim.x >= size);
-      int off = 1024 * blockIdx.x;
+      assert(gridDim.x <= 1024);
```

Comment:
in reality I think it is not limited anymore to 1024 as now I use dynamic shared

Reply:
what should be the limit ?

Reply:
if it is that `gridDim.x` be less or equal than the elements in the `extern __shared__ T psum[]` - is there a way to check it ?

Reply:
Maybe using something like https://stackoverflow.com/a/40021087/2050986?